api/src/main/java/jakarta/servlet/http/Cookie.java

```diff
@@ -422,4 +423,101 @@ public void setHttpOnly(boolean isHttpOnly) {
     public boolean isHttpOnly() {
         return isHttpOnly;
     }
+
+    /**
+     * Sets the <i>SameSite</i> flag for this cookie with the provided value.
+     *
+     * <p>
+     * When <code>sameSiteMode</code> is not <code>null</code>, this cookie will have its <code>SameSite</code>
+     * attribute set with the provided value.
+     *
+     * @param sameSiteMode the <i>SameSite</i> attribute's value; may be null if the attribute should not be set
+     *
+     * @since Servlet 4.x
+     */
+    public void setSameSiteMode(SameSiteMode sameSiteMode) {
+        this.sameSiteMode = sameSiteMode;
+    }
+
+    /**
+     * Returns the <i>SameSite</i> mode associated with this cookie.
+     *
+     * @return the <i>SameSite</i> mode associated with this cookie; may be null to indicate that the attribute should
+     *         not be set
+     *
+     * @since Servlet 4.x
+     */
+    public SameSiteMode getSameSiteMode() {
+        return sameSiteMode;
+    }
+
+    /**
+     * The available values that can be used with the cookie's <i>SameSite</i> attribute.
+     */
+    public enum SameSiteMode {
```

Comment:
I think this should just be `SameSite`  it is already scoped by `Cookie` and mode is not mentioned in the spec.  If you really feel the need to qualify it, then it is an Attribute rather than a Mode

Reply:
done